File: 2006 - 2007/Intermediate/Swedish Chef/Main.cs
// sudo apt-get update && sudo apt-get install mono-complete && sudo apt-get upgrade -y
// mcs -out:Main.exe Main.cs && mono Main.exe

using System;
using System.Text.RegularExpressions;

namespace ACSL {
    class MainClass {

        public static String getResult(String english) {
            english = english.ToLower();
            english = Regex.Replace(english, @"(?<=\W)$", " BORK BORK BORK!");
            english = Regex.Replace(english, "the", "ZEE", RegexOptions.Multiline);
            english = Regex.Replace(english, "an", "UN", RegexOptions.Multiline);
            english = Regex.Replace(english, "au", "OO", RegexOptions.Multiline);
            english = Regex.Replace(english, @"a(?!\b)", "E", RegexOptions.Multiline);
            english = Regex.Replace(english, @"ow", "OO", RegexOptions.Multiline);
            english = Regex.Replace(english, @"o", "U", RegexOptions.Multiline);
            english = Regex.Replace(english, @"ir", "UR", RegexOptions.Multiline);
            english = Regex.Replace(english, @"tion", "SHUN", RegexOptions.Multiline);
            english = Regex.Replace(english, @"(?<=\s[^i]+)i", "EE", RegexOptions.Multiline);
            english = Regex.Replace(english, @"en\b", "EE", RegexOptions.Multiline);
            english = Regex.Replace(english, @"f", " FF", RegexOptions.Multiline);
            english = Regex.Replace(english, @"e\b", "E-A", RegexOptions.Multiline);
            english = Regex.Replace(english, @"(?!\bu)u", "OO", RegexOptions.Multiline);
            english = Regex.Replace(english, @"v", "F", RegexOptions.Multiline);
            english = Regex.Replace(english, @"w", "V", RegexOptions.Multiline);
            english = english.ToUpper();
            return english;
        }

        public static void Main (string[] args) {
            string[] lines = System.IO.File.ReadAllLines(@"./Test.in");

            // Display the file contents by using a foreach loop.
            foreach (string line in lines) {
                Console.WriteLine("INPUT: " + line);
                Console.WriteLine("Output: " + getResult(line) + "\n");
            }
        }
    }
}

File: 2006 - 2007/Intermediate/Swedish Chef/Main.java
import java.io.*;

class Main {
  public static String getResult(String str) {
    return str.toLowerCase()
      .replaceAll("(?<=\\W)$", " BORK BORK BORK!")
      .replaceAll("the", "ZEE")
      .replaceAll("an", "UN")
      .replaceAll("au", "OO")
      // .replaceAll("a(?!\\w*e\\b)", "E")
      .replaceAll("a(?!\\b)", "E")
      .replaceAll("ow", "OO")
      .replaceAll("o", "U")
      .replaceAll("ir", "UR")
      .replaceAll("tion", "SHUN")
      // (?<=\s[^i]+)i regex, only works on .net and python (re package)
      .replaceFirst("(?!\\bi)i", "EE")
      // .replaceFirst("(?<!^)i", "EE")
      .replaceAll("en\\b", "EE")
      .replaceAll("f", "FF")
      .replaceAll("e\\b", "E-A")
      .replaceAll("(?!\\bu)u", "OO")
      // .replaceAll("(?<!^)u", "OO")
      .replaceAll("v", "F")
      .replaceAll("w", "V")
      .toUpperCase();
  }
  public static void main(String[] args) {
     try {
        BufferedReader reader = new BufferedReader(new FileReader("Test.in"));
        String input;
        while ((input = reader.readLine()) != null) {
          System.out.printf("Input: %s\n", input);
          System.out.printf("Output: %s\n\n", getResult(input));
        }
        reader.close();
      } catch(IOException e) {
        System.out.println(e.getMessage());
      }
  }
}
